diff --git a/drivers/gpu/drm/panel/panel-simple.c b/drivers/gpu/drm/panel/panel-simple.c
index 0019de93be1b..c4989450d598 100644
--- a/drivers/gpu/drm/panel/panel-simple.c
+++ b/drivers/gpu/drm/panel/panel-simple.c
@@ -1429,6 +1429,10 @@ static const struct panel_desc bananapi_s070wv20_ct16 = {
 	.modes = &bananapi_s070wv20_ct16_mode,
 	.num_modes = 1,
 	.bpc = 6,
+	.connector_type = DRM_MODE_CONNECTOR_DPI,
+	.bus_format = MEDIA_BUS_FMT_RGB888_1X24,
+	.bus_flags = DRM_BUS_FLAG_DE_HIGH |
+		   DRM_BUS_FLAG_PIXDATA_SAMPLE_POSEDGE,
 	.size = {
 		.width = 154,
 		.height = 86,
